### PR TITLE
Make sure struct-fields are always sorted

### PR DESCRIPTION
**Pull request by segeljakt**

### CODE REVIEW

File: arc-script/arc-script-test/compile/src/snapshots/hir@structs.arc.snap [hunk 1]
```diff
@@ -5,7 +5,7 @@ input_file: arc-script-test/compile/src/tests/expect_pass/structs.arc
 
 ---
 fun test() -> unit {
-    let x_0: { a: { c: i32, b: i32 }, xyz: i32, d: { b: i32, c: i32 } } = { a: { c: 1, b: 5 }, xyz: 2, d: { b: 4, c: 2 } } in
+    let x_0: { a: { b: i32, c: i32 }, d: { b: i32, c: i32 }, xyz: i32 } = { a: { b: 5, c: 1 }, d: { b: 4, c: 2 }, xyz: 2 } in
```
Comment (frej) [inline arc-script/arc-script-test/compile/src/snapshots/hir@structs.arc.snap:8]: Is this safe to do? What if the code was:

```
fun foo(a: {c: i32, b: i32}, b: {b:i32, c:i32}) ->
    {a:{c:i32, b:i32}, xyz:i32, d:{b:i32, c:i32}} {
    let r = {a:a, xyz:get_counter(), d:get_counter()} in
    r
}
```

where `get_counter()` has a side effect? Have we made / should we make any promises about the evaluation order of initializers?

**Review by frej — COMMENTED** [review]
See the line comment about evaluation order.

File: arc-script/arc-script-test/compile/src/snapshots/hir@structs.arc.snap (hunk 1 above)
Comment (segeljakt) [reply arc-script/arc-script-test/compile/src/snapshots/hir@structs.arc.snap]: Good point, fixed now!

Comment (segeljakt) [reply arc-script/arc-script-test/compile/src/snapshots/hir@structs.arc.snap]: I also ran `cargo fmt` which changed some other files, sorry 👀

**Review by frej — APPROVED** [review]
LGTM

**Comment by segeljakt** [comment]
One minor update, I removed the code in HIR => Arcon for sorting fields, so now all fields are sorted before then.

**Comment by frej** [comment]
Go ahead and merge it!

**Pull request merged**